test/unit/overlay.js

```diff
@@ -595,4 +595,17 @@ describe('Overlays', function () {
         fixtures.assertSimilar(fixtures.expected('overlay-jpeg-with-jpeg.jpg'), data, done);
       });
   });
+
+  it('Overlays PNG on a solid background using difference blend mode', function (done) {
```

Comment:
It's always a pleasure to see tests in a PR, thank you. Please can you also add tests for the error handling that's been introduced around invalid `blendMode` values.